File: java_solutions_1/435.non-overlapping-intervals.java
/*
 * @lc app=leetcode id=435 lang=java
 *
 * [435] Non-overlapping Intervals
 */

class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        return intervals.length == 0 ? 0 : java.util.stream.Stream.of(0).peek(_v -> java.util.Arrays.sort(intervals, (a, b) -> Integer.compare(a[1], b[1]))).map(_v -> java.util.Arrays.stream(intervals).skip(1).reduce(new int[]{intervals[0][1], 0}, (state, next) -> next[0] < state[0] ? new int[]{state[0], state[1] + 1} : new int[]{next[1], state[1]}, (a, b) -> b)[1]).findFirst().get();
    }
}

File: kotlin_solutions/435.non-overlapping-intervals.kt
/*
 * @lc app=leetcode id=435 lang=kotlin
 *
 * [435] Non-overlapping Intervals
 */

class Solution {fun eraseOverlapIntervals(intervals: Array<IntArray>): Int = intervals.sortedBy { it[1] }.fold(-50001 to 0) { (end, cnt), interval -> if (interval[0] >= end) interval[1] to cnt else end to cnt + 1 }.second}

